src/main/java/uk/gov/ida/verifyserviceprovider/configuration/HubMetadataConfiguration.java

```diff
@@ -108,4 +110,22 @@ private KeyStore validateTruststore(KeyStore trustStore) {
         }
         return trustStore;
     }
+
+    private KeyStore tryGetTrustStore(TrustStoreConfiguration trustStoreConfiguration) {
+        try {
+            return trustStoreConfiguration.getTrustStore();
+        } catch (Exception ignored) {
```

Comment:
What exception are you expecting to catch here, and why is it ignored? Also, seems odd that null is being returned.

Reply:
Actually, the null thing isn't that odd, as it's being returned to `map()` which gives you an empty optional.

Reply:
It depends on the specific implementation - for instance for the `FileBackedTrustStoreConfiguration` it will attempt to load filestore even if the path is null - it doesn't do any checks. Those implementations live in saml-libs and are used in other places. I guess a "proper" fix would be to add check to the classes in saml-libs instead

Reply:
For instance here you'd get a NullPointerException
https://github.com/alphagov/verify-saml-libs/blob/c97880d1becb816d5b5a0d41a62ddf0da208e298/saml-lib/src/main/java/uk/gov/ida/saml/metadata/FileBackedTrustStoreConfiguration.java#L21

Reply:
OK actually the only case we want to ignore is NPE so I've changed this to only ignore that - this way if a value is provided but it's invalid it will still show up in the log and prevent the app from starting 👍